mixnode/src/commands/run.rs

```diff
@@ -4,56 +4,52 @@
-                .takes_value(true)
-                .required(true),
-        )
-        // the rest of arguments are optional, they are used to override settings in config file
-        .arg(
-            Arg::with_name(HOST_ARG_NAME)
-                .long(HOST_ARG_NAME)
-                .help("The custom host on which the mixnode will be running")
-                .takes_value(true),
-        )
-        .arg(
-            Arg::with_name(MIX_PORT_ARG_NAME)
-                .long(MIX_PORT_ARG_NAME)
-                .help("The port on which the mixnode will be listening for mix packets")
-                .takes_value(true),
-        )
-        .arg(
-            Arg::with_name(VERLOC_PORT_ARG_NAME)
-                .long(VERLOC_PORT_ARG_NAME)
-                .help("The port on which the mixnode will be listening for verloc packets")
-                .takes_value(true),
-        )
-        .arg(
-            Arg::with_name(HTTP_API_PORT_ARG_NAME)
-                .long(HTTP_API_PORT_ARG_NAME)
-                .help("The port on which the mixnode will be listening for http requests")
-                .takes_value(true),
-        )
-        .arg(
-            Arg::with_name(ANNOUNCE_HOST_ARG_NAME)
-                .long(ANNOUNCE_HOST_ARG_NAME)
-                .help("The host that will be reported to the directory server")
-                .takes_value(true),
-        )
-        .arg(
-            Arg::with_name(VALIDATORS_ARG_NAME)
-                .long(VALIDATORS_ARG_NAME)
-                .help("Comma separated list of rest endpoints of the validators")
-                .takes_value(true),
-        )
+#[derive(Args, Clone)]
+pub(crate) struct Run {
+    /// Id of the nym-mixnode we want to run
+    #[clap(long)]
+    id: String,
+
+    /// The custom host on which the mixnode will be running
+    #[clap(long)]
+    host: Option<String>,
+
+    /// The port on which the mixnode will be listening for mix packets
+    #[clap(long)]
+    mix_port: Option<u16>,
+
+    /// The port on which the mixnode will be listening for verloc packets
+    #[clap(long)]
+    verloc_port: Option<u16>,
+
+    /// The port on which the mixnode will be listening for http requests
+    #[clap(long)]
+    http_api_port: Option<u16>,
+
+    /// The host that will be reported to the directory server
+    #[clap(long)]
+    announce_host: Option<String>,
+
+    /// Comma separated list of rest endpoints of the validators
+    #[clap(long)]
+    validators: Option<String>,
+}
+
+impl From<Run> for OverrideConfig {
+    fn from(run_config: Run) -> Self {
+        OverrideConfig {
+            id: run_config.id,
+            host: run_config.host,
+            mix_port: run_config.mix_port,
+            verloc_port: run_config.verloc_port,
+            http_api_port: run_config.http_api_port,
+            announce_host: run_config.announce_host,
+            validators: run_config.validators,
+            wallet_address: None,
```

Comment:
I think this might be an omission from the previous code, because if you can set a particular value during `init`, you should be able to override it during `run` and hence the `wallet_address` field should also exist here.